% !TEX TS-program = pdflatex
% !TEX encoding = UTF-8 Unicode

\documentclass[12pt,oneside,letterpaper]{article}

% release codenumber (update it here!)
\newcommand\releasecode{0.2.0}

%%% PACKAGES
\usepackage{lmodern} % Activa fuentes y macros de Latin Modern
\usepackage[T1]{fontenc} % set font encoding
\usepackage[utf8]{inputenc} % set input encoding (not needed with XeLaTeX)
\usepackage[english]{babel} 
\usepackage[expert, charter]{mathdesign} % Fuentes... disponibles conjuntos {charter|utopia}
%%\usepackage{calligra}
\usepackage[printwatermark]{xwatermark}
\usepackage{graphicx}
\usepackage[usenames,dvipsnames,table]{xcolor}
\usepackage{parskip}
%%\usepackage{float}
%\usepackage[nolists,tablesfirst, nomarkers]{endfloat}
\usepackage{moreverb} % for verbatim output

% text layout
\usepackage[letterpaper]{geometry}
\geometry{textwidth=16.25cm} % 15.25cm for single-space, 16.25cm for double-space
\geometry{textheight=22.5cm} % 22cm for single-space, 22.5cm for double-space

% watermarks (time consuming!)
\newwatermark[allpages,color=blue!4,angle=45,scale=9,xpos=-40,ypos=0]{DRAFT}
\newwatermark[allpages,color=blue!12,angle=0,scale=1,xpos=0,ypos=137]{
\emph{mSystems} \LaTeX\ manuscript release \releasecode}

% helps to keep figures from being orphaned on a page by themselves
\renewcommand{\topfraction}{0.85}
\renewcommand{\textfraction}{0.1}
%% Roman footnote numbering style to not collide with bibliographic cites: 
\renewcommand{\thefootnote}{\Roman{footnote}}

% line numbering
\usepackage[running,mathlines]{lineno}
\renewcommand\thelinenumber{\color{red}\arabic{linenumber}}
\linenumbers

% bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,font=small]{caption}

% review layout with double-spacing
\usepackage{setspace} 
\captionsetup{labelfont=bf,labelsep=period,font=doublespacing}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}
\renewcommand\citeleft{(}
\renewcommand\citeright{)}
\renewcommand\citeform[1]{\textsl{#1}}

% Remove brackets from numbering in list of References
\renewcommand\refname{\large References}
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

% Importance environment and keywords command
\newcommand\importancename{Importance}
\newcommand{\importance}[1]{
	{
    \small
    \begin{center}
        {\bfseries \importancename\vspace{-.5em}\vspace{0pt}}
    \end{center}
    \begin{quote}
    #1
    \end{quote}
    }
}
\providecommand{\keywords}[1]{\footnotesize\textbf{\textit{Keywords---}} #1}

% Package authblk
\usepackage{authblk}
\renewcommand\Authands{ \& }
\renewcommand\Authfont{\normalsize \bf}
\renewcommand\Affilfont{\small \normalfont}

% notation
\usepackage{amsmath}

%% Roman footnote numbering style to not collide with bibliographic cites: 
\renewcommand{\thefootnote}{\Roman{footnote}}

% aux commands
\newcommand{\CC}[0]{\emph{cmplxcruncher}}
\newcommand{\task}[1]{\texttt{\bfseries\scshape\textcolor{MidnightBlue}{#1}}}
\newcommand{\un}[1]{\operatorname{#1}}
\newcommand{\unclassrate}[0]{\un{kpb/s/core}}
\newcommand{\todo}[1]{\texttt{\bfseries\textcolor{Orange}{#1}}}


% word count (Need --enable-write18 or --shell-escape)
\immediate\write18{texcount -opt=option.tc \jobname.tex > wordcount.aux} 
\newcommand\wordcount{\verbatiminput{wordcount.aux}}


%%% DOCUMENT %%%

\begin{document}

\title{
	\vspace*{0mm} % May help to include the corresponding author footnote in the title page
	\singlespacing
	\begin{flushleft}
		\texttt{\large Title:} \\
	\end{flushleft}
	\vspace*{2mm}
	Health and disease imprinted in the time variability of the human microbiome\\
	\vspace*{6mm}
	\begin{flushleft}
		\texttt{\large Running title:} \\
	\end{flushleft}
	\vspace*{0mm}
	Microbiota, are you sick?
	\vspace*{4mm}
	}

\doublespacing

\author[1,2,$*$]{Jose Manuel Martí}
\author[1,2,3,$*$]{Daniel Martínez-Martínez}
\author[2]{Manuel Peña}
\author[1,2]{César Gracia}
\author[1,3,4,5]{Amparo Latorre}
\author[1,3,4,5]{Andrés Moya}
\author[1,2,\#]{Carlos P. Garay}

\affil[1]{Institute for Integrative Systems Biology (I2SysBio), 46980, Spain.}
\affil[2]{Instituto de Fisica Corpuscular, CSIC-UVEG, P.O.  22085, 46071, Valencia, Spain.}
\affil[3]{FISABIO, Avda de Catalunya, 21, 46020, Valencia, Spain.}
\affil[4]{Cavanilles Institute of Biodiversity and Evolutionary Biology, UVEG, 46980, Spain.}
\affil[5]{CIBER en Epidemiología y Salud Pública (CIBEResp), Madrid, Spain}

\date{}

\maketitle
\wordcount
\footnote[0]{$^*$ Equally contributed}
\footnote[0]{$^\#$ Corresponding author: penagaray@gmail.com}

\clearpage

%TC:newcounter abswords Words in abstract
%TC:macro \abstract [abswords]
{\abstract{Animal microbiota (human included) plays an important role keeping healthy the physiological status of the host. Increasing research activity is dedicated to understand how changes in composition and function of the microbiota are associated to disease or not. We analyze 16S rRNA and whole genome sequencing (WGS) published data from the gut microbiota of 97 individuals monitored in time. Temporal fluctuations in the microbial composition reveal significant differences due to factors such us dietary changes, antibiotic intake, age or disease. Here we show that a fluctuation scaling law describes the temporal changes in the gut microbiota. This law allows to estimate the temporal variability of the microbial population and quantitatively characterizes the path toward disease by a noise-induced phase transition. The estimation of the systemic parameters for follow-up studies may have clinical use and, more generally, applications in other fields where it is important to know if a given community is stable or not.}}

%TC:newcounter impwords Words in importance section
%TC:macro \importance [impwords]
\importance{Human microbiota is tightly associated to the health status of a person. Here we analyse the microbial composition of several subjects under different conditions, over a time span that ranges from days to months. Using the Langevin equation as the basis of our mathematical framework in order to evaluate microbial temporal stability, we prove that we are capable to distinguish stable from unstable microbiotas. This first step will help us to determine how microbiota temporal stability is related to the healthiness of the people, and it will allow the development of a more complete framework in order to deepen the knowledge of this complex system.}

\vspace{4mm}
\begin{keywords}
microbiome, systems biology, ecological modeling, metagenomics, stability	
\end{keywords}

%===================================== SECTIONS =======================================
%%% Introduction
\section*{Introduction}
\input{mSys_intro.tex}
\clearpage
%%% Result
\section*{Results}
\input{mSys_results.tex}
\clearpage
%%% Discussion
\section*{Discussion}
\input{mSys_discussion.tex}
\clearpage    
%%% Material & Methods
\section*{Materials and Methods}
\input{mSys_MandM.tex}   
\clearpage

\section*{Acknowledgments}
Authors declare that there are no competing financial interests in relation to the work described here. 

\section*{Funding Information}
This work is supported by Generalitat Valenciana Prometeo Grants II/2014/050, II/2014/065, by the Spanish Grants FPA2011-29678, BFU2012-39816-C02-01 of MINECO and by PITN-GA-2011-289442-INVISIBLES. JMM \& DMM acknowledge FPI and FISABIO fellowships. \todo{Modificar becas de JMM y DMM ¿poner algún grant más?}

\clearpage
\begin{thebibliography}{99}
\bibitem{bileacids} {\bf Swann JR, Want EJ, Geier FM, Spagou K, Wilson ID, Sidaway JE, Nicholson JK, Holmes E.} 2011. Systemic gut microbial modulation of bile acid metabolism in host tissue compartments. Proc Natl Acad Sci {\bf 108}:4523–4530.
\bibitem{choline} {\bf nSpencer MD, Hamp TJ, Reid RW, Fischer LM, Zeisel SH, Fodor AA.} 2011. Association between composition of the human gastrointestinal microbiome and development of fatty liver with choline deficiency. Gastroenterology {\bf 140}:976–986.
\bibitem{scfa1} {\bf Samuel BS, Shaito A, Motoike T, Rey FE, Backhed F, Manchester JK, Hammer RE, Williams SC, Crowley J, Yanagisawa M, Gordon JI}. 2008. Effects of the gut microbiota on host adiposity are modulated by the short-chain fatty-acid binding G protein-coupled receptor, Gpr41. Proc Natl Acad Sci {\bf 105}:16767–16772.
\bibitem{scfa2} {\bf Smith PM, Howitt MR, Panikov N, Michaud M, Gallini CA, Bohlooly-Y M, Glickman JN, Garrett WS.} 2013. The Microbial Metabolites, Short-Chain Fatty Acids, Regulate Colonic Treg Cell Homeostasis. Science (80- ) {\bf 341}:569–573.
\bibitem {scfa3} {\bf Kimura I, Ozawa K, Inoue D, Imamura T, Kimura K, Maeda T, Terasawa K, Kashihara D, Hirano K, Tani T, Takahashi T, Miyauchi S, Shioi G, Inoue H, Tsujimoto G.} 2013. The gut microbiota suppresses insulin-mediated fat accumulation via the short-chain fatty acid receptor GPR43. Nat Commun {\bf 4}:1829.
\bibitem {scfa4} {\bf Maslowski KM, Vieira AT, Ng A, Kranich J, Sierro F, Di Yu, Schilter HC, Rolph MS, Mackay F, Artis D, Xavier RJ, Teixeira MM, Mackay CR.} 2009. Regulation of inflammatory responses by gut microbiota and chemoattractant receptor GPR43. Nature {\bf 461}:1282–1286.
\bibitem{diabetes2} {\bf Qin J, Li Y, Cai Z, Li S, Zhu J, Zhang F, Liang S, Zhang W, Guan Y, Shen D, Peng Y, Zhang D, Jie Z, Wu W, Qin Y, Xue W, Li J, Han L, Lu D, Wu P, Dai Y, Sun X, Li Z, Tang A, Zhong S, Li X, Chen W, Xu R, Wang M, Feng Q, Gong M, Yu J, Zhang Y, Zhang M, Hansen T, Sanchez G, Raes J, Falony G, Okuda S, Almeida M, LeChatelier E, Renault P, Pons N, Batto J-M, Zhang Z, Chen H, Yang R, Zheng W, Li S, Yang H, Wang J, Ehrlich SD, Nielsen R, Pedersen O, Kristiansen K, Wang J.} 2012. A metagenome-wide association study of gut microbiota in type 2 diabetes. Nature {\bf 490}:55–60.
\bibitem{CVD} {\bf Brown JM, Hazen SL.} 2015. The Gut Microbial Endocrine Organ: Bacterially Derived Signals Driving Cardiometabolic Diseases. Annu Rev Med {\bf 66}:343–359.
\bibitem{IBS} {\bf Durbán A, Abellán JJ, Jiménez-Hernández N, Artacho A, Garrigues V, Ortiz V, Ponce J, Latorre A, Moya A.} 2013. Instability of the faecal microbiota in diarrhoea-predominant irritable bowel syndrome. FEMS Microbiol Ecol {\bf 86}:581–589.
\bibitem{CD} {\bf Gevers D, Kugathasan S, Denson LA, Vázquez-Baeza Y, Van Treuren W, Ren B, Schwager E, Knights D, Song SJ, Yassour M, Morgan XC, Kostic AD, Luo C, González A, McDonald D, Haberman Y, Walters T, Baker S, Rosh J, Stephens M, Heyman M, Markowitz J, Baldassano R, Griffiths A, Sylvester F, Mack D, Kim S, Crandall W, Hyams J, Huttenhower C, Knight R, Xavier RJ.} 2014. The treatment-naive microbiome in new-onset Crohn’s disease. Cell Host Microbe {\bf 15}:382–392.
\bibitem{ob1} {\bf Ridaura VK, Faith JJ, Rey FE, Cheng J, Duncan AE, Kau L, Griffi NW, Lombard V, Henrissat B, Bain JR, Michael J, Ilkayeva O, Semenkovich CF, Funai K, Hayashi DK, Lyle J, Martini MC, Ursell LK, Clemente JC, Treuren W Van, William A, Knight R, Newgard CB, Heath AC, Gordon JI, Kau AL, Griffin NW, Muehlbauer MJ.} 2013. Gut Microbiota from Twins Discordant for Obesity Modulate Metabolism in Mice Gut Microbiota from Twins Metabolism in Mice. Science {\bf 341}:1241214.
\bibitem{ob2} {\bf Turnbaugh PJ, Hamady M, Yatsunenko T, Cantarel BL, Duncan A, Ley RE, Sogin ML, Jones WJ, Roe BA, Affourtit JP, Egholm M, Henrissat B, Heath AC, Knight R, Gordon JI.} 2009. LETTERS A core gut microbiome in obese and lean twins. Nature {\bf 457}:480–484.
\bibitem{nutr} {\bf Subramanian S, Huq S, Yatsunenko T, Haque R, Mahfuz M, Alam MA, Benezra A, DeStefano J, Meier MF, Muegge BD, Barratt MJ, VanArendonk LG, Zhang Q, Province MA, Petri WA, Ahmed T, Gordon JI.} 2014. Persistent gut microbiota immaturity in malnourished Bangladeshi children. Nature {\bf 510}:417–21.
\bibitem{Moya_trends} {\bf Moya A, Ferrer M.} 2016. Functional Redundancy-Induced Stability of Gut Microbiota Subjected to Disturbance. Trends Microbiol {\bf 24}:402–413.
\bibitem{microb&health} {\bf Marchesi JR, Adams DH, Fava F, Hermes GD a, Hirschfield GM, Hold G, Quraishi MN, Kinross J, Smidt H, Tuohy KM, Thomas L V, Zoetendal EG, Hart A.} 2015. The gut microbiota and host health: a new clinical frontier. Gut 1–10.
\bibitem{normal1} {\bf Falony G, Joossens M, Vieira-Silva S, Wang J, Darzi Y, Faust K, Kurilshikov A, Bonder MJ, Valles-Colomer M, Vandeputte D, Tito RY, Chaffron S, Rymenans L, Verspecht C, De Sutter L, Lima-Mendez G, Dhoe K, Jonckheere K, Homola D, Garcia R, Tigchelaar EF, Eeckhaudt L, Fu J, Henckaerts L, Zhernakova A, Wijmenga C, Raes J.} 2016. Population-level analysis of gut microbiome variation. Science (80- ) {\bf 352}:560–564.
\bibitem{normal2} {\bf Zhernakova A, Kurilshikov A, Bonder MJ, Tigchelaar EF, Schirmer M, Vatanen T, Mujagic Z, Vila AV, Falony G, Vieira-Silva S, Wang J, Imhann F, Brandsma E, Jankipersadsing SA, Joossens M, Cenit MC, Deelen P, Swertz MA, Weersma RK, Feskens EJM, Netea MG, Gevers D, Jonkers D, Franke L, Aulchenko YS, Huttenhower C, Raes J, Hofker MH, Xavier RJ, Wijmenga C, Fu J.} 2016. Population-based metagenomics analysis reveals markers for gut microbiome composition and diversity. Science (80- ) {\bf 352}:565–569.
\bibitem{sysbio&microb} {\bf Wu H, Tremaroli V, Bäckhed F.} 2015. Linking Microbiota to Human Diseases: A Systems Biology Perspective. Trends Endocrinol Metab {\bf 26}:758–770.
\bibitem{msys1} {\bf Noecker C, Eng A, Srinivasan S, Theriot CM, Young VB, Jansson JK, Fredricks DN, Borenstein E.} 2016. Metabolic Model-Based Integration of Microbiome Taxonomic and Metabolomic Profiles Elucidates Mechanistic Links between Ecological and Metabolic Variation. mSystems {\bf 1}:e00013–15.
\bibitem {metasysbio} {\bf Greenblum S, Turnbaugh PJ, Borenstein E.} 2012. Metagenomic systems biology of the human gut microbiome reveals topological shifts associated with obesity and inflammatory bowel disease. Proc Natl Acad Sci {\bf 109}:594–599.
\bibitem{taylor} {\bf Taylor, L.R.} 1961. Aggregation, Variance and the mean. Nature {\bf 189,} 732-35.
\bibitem{randomwalks} {\bf de Menezes MA, Barabási A-L.} 2004. Fluctuations in network dynamics. Phys Rev Lett {\bf 92}:1–4.
\bibitem{economics1} {\bf Mantegna RN, Stanley HE.} 1995. Scaling behaviour in the dynamics of an economic index. Nature {\bf 376}:46–49.
\bibitem{economics2} {\bf Eisler Z, Kertesz J, Yook SH, Barabasi AL.} 2005. Multiscaling and non-universality in fluctuations of driven complex systems. Europhys Lett {\bf 69}:664–670.
\bibitem{animal1} {\bf Reed DH, Hobbs GR.} 2004. The relationship between population size and temporal variability in population size. Anim Conserv {\bf 7}:1–8.
\bibitem{animal2} {\bf Anderson RM, Gordon DM, Crawley MJ, Hassell MP.} 1982. Variability in the abundance of animal and plant species. Nature {\bf 18}: 245–248
\bibitem{genexpress} {\bf Živković J, Tadić B, Wick N, Thurner S.} 2006. Statistical indicators of collective behavior and functional clusters in gene networks of yeast. Eur Phys J B {\bf 50}:255–258.
\bibitem{genome} {\bf Kendal WS.} 2003. An Exponential Dispersion Model for the Distribution of Human Single Nucleotide Polymorphisms. Mol Biol Evol {\bf 20}:579–590.
\bibitem{isme1} {\bf Zhang Z, Geng J, Tang X, Fan H, Xu J, Wen X, Ma ZS, Shi P.} 2014. Spatial heterogeneity and co-occurrence patterns of human mucosal-associated intestinal microbiota. ISME J {\bf 8}:881–93.
\bibitem{cobas} {\bf Pérez-Cobas AE, Artacho A, Ott SJ, Moya A, Gosalbes MJ, Latorre A.} 2014. Structural and functional changes in the gut microbiota associated to Clostridium difficile infection. Front Microbiol {\bf 5}:1–15.
\bibitem{schloss} {\bf Ding T, Schloss PD.} 2014. Dynamics and associations of microbial community types across the human body. Nature {\bf 509}:357–360.
\bibitem{ravel} {\bf Gajer P, Brotman RM, Bai G, Sakamoto J, Schütte UME, Zhong X, Koenig SSK, Fu L, Ma ZS, Zhou X, Abdo Z, Forney LJ, Ravel J.} 2012. Temporal dynamics of the human vaginal microbiota. Sci Transl Med {\bf 4}:132ra52.
\bibitem{ranking} {\bf Blumm N, Ghoshal G, Forró Z, Schich M, Bianconi G, Bouchaud J-P, Barabási A-L.} 2012. Dynamics of Ranking Processes in Complex Systems. Phys Rev Lett {\bf 109}:128701.
\bibitem{moving} Caporaso, J.G. et al. Moving pictures of the human microbiome. {\it Genome Biol.} {\bf 12,} R50 (2011).
\bibitem{LEA} Faith, J.J. et al. The long-term stability of the human gut microbiota. {\it Science} {\bf 341,} 1237439 (2013).
\bibitem{kwashiorkor} Smith M.I. et al. Gut microbiomes of Malawian twin pairs discordant for kwashiorkor. {\it Science} {\bf 339,} 548-54 (2013).
\bibitem{diet} David, L.A. et al. Diet rapidly and reproducibly alters the human gut microbiome. {\it Nature} {\bf 505,} 559-63 (2014).
\bibitem{antibiotic} Dethlefsen L., Relman D. A. Incomplete recovery and individualized responses of the human distal gut microbiota to repeated antibiotic perturbation. {\it Proc. Nat. Acad. Sci. USA} {\bf 108,} 4554-61 (2011).
\bibitem{QIIME} Caporaso, J.G. et al. QIIME allows analysis of high-throughput community sequencing data. {\it Nature Methods} {\bf 7,} 335-6 (2010).
\bibitem{LMAT} {\bf Ames SK, Hysom DA, Gardner SN, Lloyd GS, Gokhale MB, Allen JE}.  2013. Scalable metagenomic taxonomy classification using a reference genome database.  Bioinformatics {\bf 29}:2253-2260.
\bibitem{LMAT2} {\bf Ames SK, Gardner SN, Marti JM, Slezak TR, Gokhale MB, Allen JE}. 2015. Using populations of human and microbial genomes for organism detection in metagenomes. Genome Res. {\bf 25}:1056-67.
\bibitem{fs} Eisler,Z., Bartos,I., Kertesz,J. Fluctuation scaling in complex systems: Taylor's law and beyond. {\it Adv. Phys.} {\bf 57,} 85 (2008).
\bibitem{stat} Jorgensen,B., Martinez,J.R., Tsao,M. Asymptotic behaviour of the variance function. {\it Scand. J. Statist..} {\bf 21,} 223-243 (1994).
\bibitem{convergence1} Fronczak,A., Fronczak,P. Origins of Taylor's power law for fluctuation scaling in complex systems. {\it Phys. Rev. E} {\bf 81,} 066112 (2010).
\bibitem{convergence2} Kendal, W.S., Jorgensen,B. Taylor's power law and fluctuation scaling explained by a central-limit-like convergence. {\it Phys. Rev. E} {\bf 83,} 066115 (2011).
\bibitem{convergence3} Kendal, W.S., Jorgensen,B. Tweedie convergence: A mathematical basis for Taylor's power law. {\it Phys. Rev. E} {\bf 84,} 066120 (2011).

	\bibitem{FD} Weber, J. \textit{et al.} Fluctuation dissipation theorem. {\it Phys. Rev.} {\bf 101}, 1620-6 (1956).
	\bibitem{FASTX} Gordon, A., Hannon, G.J. FASTX-Toolkit. FASTQ/A shortreads pre-processing tools (2010). http://hannonlab.cshl.edu/fastx\_toolkit/ (accessed 23 Feb 2015).
	\bibitem{SILVA} Quast C.  \textit{et al.} The SILVA ribosomal RNA gene database project: improved data processing and web-based tools (2013)
 	\bibitem{ecology} Xiao Xiao, Ethan P. White, Mevin B. Hooten, and Susan L. Durham. On the use of log-transformation vs. nonlinear regression for analyzing biological power laws. {\it Ecology} {\bf 92}, 10, 1887-1894 (2011).
 	\bibitem{genR2} Magee L., $R^2$ measures based on wald and likelihood ratio joint significance tests. {\it The American Statistician} {\bf 44}, 3, 250-253 (1990).
	\bibitem{disR2} Nagelkerke N.J.D., A note on a general definition of the coefficient of determination. {\it Biometrika} {\bf 78}, 3, 691-692 (1991).
	\bibitem{boot} Wu, C.F.J. Jackknife, bootstrap and other resampling methods in regression analysis. (with discussions) \textit{The Annals of Statistics} {\bf 14}: 1261?1350 (1986)
\end{thebibliography}
\todo{Eliminar \textit{et al.} y poner la referencia completa como exige la guía de estilo de la revista...}

\end{document}


